tests/e2e/lib.sh

```diff
@@ -665,6 +684,60 @@ check_for_stackrox_OOMs() {
     fi
 }
 
+junit_report_pod_restarts() {
+    local check_output="${1:-}"
+
+    local previous_logs=()
+    while IFS='' read -r line; do previous_logs+=("$line"); done < <(echo "${check_output}" | grep "copied to Artifacts" | cut -d" " -f1 | sort -u)
+
+    local previous_log_regex=""
+    declare -A found_previous_logs
+    declare -A found_pod_container_keys
+    for previous_log in "${previous_logs[@]}"
+    do
+        found_previous_logs["${previous_log}"]=""
+        for map_key in "${!POD_CONTAINERS_MAP[@]}"
+        do
+            previous_log_regex="${POD_CONTAINERS_MAP[${map_key}]}"
+            if [[ "${previous_log}" =~ $previous_log_regex ]]; then
+                found_previous_logs["${previous_log}"]="${map_key}"
+                found_pod_container_keys["${map_key}"]="found"
+                break
+            fi
+        done
+    done
+
+    # (FAILURES - Fallback) Report failed, but not found pods in defined pod->container map.
+    local crop_pod_name=""
+    for previous_log in "${!found_previous_logs[@]}"
+    do
+        if [[ "${found_previous_logs[${previous_log}]}" != "" ]]; then
+            continue
+        fi
+
+        crop_pod_name="$(echo "${previous_log}" | cut -d- -f1)"
+        save_junit_failure "Pod Restarts" "${crop_pod_name}" "${check_output}"
+    done
+
+    # (FAILURES) Report failed and found pods. We use improved test name matching.
+    for map_key in "${!found_pod_container_keys[@]}"
+    do
+        save_junit_failure "Pod Restarts" "${map_key}" "${check_output}"
+    done
+
+    # Report pods without restarts (SUCCESSES).
+    local found_map_key=""
+    for map_key in "${!POD_CONTAINERS_MAP[@]}"
+    do
+        found_map_key="${found_pod_container_keys[${map_key}]:-}"
+        if [[ "${found_map_key}" == "found" ]]; then
+            continue
+        fi
+
+        save_junit_success "Pod Restarts" "${map_key}"
```

Comment:
This sounds a bit strange 😅 
```
$ junit_report_pod_restarts 
Success: Pod Restarts pod: central-db - container: init-db
Success: Pod Restarts pod: collector - container: node-inventory
Success: Pod Restarts pod: collector - container: collector
(...)
```
it looks like we would want the pods to restart and that operation was successful. 

Maybe we could use something like:
```suggestion
        save_junit_success "Check unexpected pod restarts" "${map_key}"
```
then status success and failure would make sense. WDYT?

Reply:
Thanks for improoving!